### PR TITLE
Add x-checker-data

### PR DESCRIPTION
**Pull request by proletarius101**
- Autoupdate: add x-checker-data

- `evince` and `gnome-desktop` are slightly updated as a side effect - I need to test the latest version the `x-checker-data` will propagate to

### CODE REVIEW

**Comment by proletarius101** [comment]
@gpoo

File: org.gnome.Evince.json
Comment (gpoo) [inline org.gnome.Evince.json:239]: This should be the stable version. Not the development one. Users who wants to test the development version should install it from gnome-nightly (or whatever name it is currently).

File: org.gnome.Evince.json
Comment (gpoo) [inline org.gnome.Evince.json:233]: This flags should be a separate PR, or at least an independent commit. That said, the stable version does not need to match the development one. In the development version we skip the documentation because it runs in the CI after every commit, and we want to make it fast. This is for stable distribution of Evince in flathub.

File: org.gnome.Evince.json
Comment (gpoo) [inline org.gnome.Evince.json:160]: This should use the stable version. Not the development one.

File: org.gnome.Evince.json
Comment (gpoo) [inline org.gnome.Evince.json:22]: Thank you for the PR.

This change has nothing to do with adding x-checker-data. Please remove it from this PR.

File: org.gnome.Evince.json
Comment (gpoo) [inline org.gnome.Evince.json:144]: why do we need to use the git version and not the released one? I prefer the former.

File: org.gnome.Evince.json
Comment (gpoo) [inline org.gnome.Evince.json:188]: Same here.

File: org.gnome.Evince.json
Comment (proletarius101) [reply org.gnome.Evince.json:239]: Well, isn't it the latest stable version (Release)? At least they tagged it this way, so as their https://gitlab.gnome.org/GNOME/evince/-/blob/master/org.gnome.Evince.appdata.xml.in

File: org.gnome.Evince.json
Comment (proletarius101) [reply org.gnome.Evince.json:233]: Then I'll revert this one

File: org.gnome.Evince.json
Comment (proletarius101) [reply org.gnome.Evince.json:160]: Ditto. Why is it the development version when they tagged it ? Or maybe they have a different release scheme?

**Comment by proletarius101** [comment]
Interesting. Why do you think I'm putting the development ones? They are simply hosted on GitLab, but they have the same version (as tags) as the ones hosted on the website. I don't use the one on download.gnome.org because the url doesn't compatible with the x-checker-data. e.g.
````
                     "url": "https://download.gnome.org/sources/gspell/1.8/gspell-1.8.4.tar.xz", 
````
There is a `1.8` in the middle. But what if the version bumps to `1.9.x`? We don't have a procedure to provide the substring `1.9`.

File: org.gnome.Evince.json
Comment (hadess) [inline org.gnome.Evince.json:15]: Why did you remove this?

File: org.gnome.Evince.json
Comment (hadess) [inline org.gnome.Evince.json:15]: The DConf migration should be a separate commit.

File: org.gnome.Evince.json
Comment (hadess) [inline org.gnome.Evince.json:199]: Adding libhandy should be a separate commit too.

File: org.gnome.Evince.json
Comment (hadess) [reply org.gnome.Evince.json:237]: Odd minor releases are development versions. 3.39.X releases are all development releases for version 3.40 (which will actually be called version 40). This isn't a stable version.

Not sure why you would think that German, who's one of the upstream maintainers for evince would get that wrong.

**Review by hadess — CHANGES_REQUESTED** [review]

**Comment by proletarius101** [comment]
> Odd minor releases are development versions. 3.39.X releases are all development releases for version 3.40 (which will actually be called version 40). This isn't a stable version.

That could be avoided in the flatpak side definitely. How about other libs? Is it the GNOME versioning scheme which applies to all GNOME projects?

> Not sure why you would think that German, who's one of the upstream maintainers for evince would get that wrong.

Well, not aware of it

File: org.gnome.Evince.json
Comment (proletarius101) [reply org.gnome.Evince.json:199]: It's a dependency of newer versions of Evince... How can I update Evince without adding it, or adding it without bump the version which is useless for the old version?

File: org.gnome.Evince.json
Comment (proletarius101) [reply org.gnome.Evince.json:15]: Would it even work if I bump the version without this line?

Comment (proletarius101) [reply org.gnome.Evince.json:15]: Oh that's a redundent one

File: org.gnome.Evince.json
Comment (bilelmoussaoui) [reply org.gnome.Evince.json:161]: The gnome modules should definitely wait until the gnome checker lands otherwise you will have to deal with the stable/instable release schemes.

File: org.gnome.Evince.json
Comment (bilelmoussaoui) [reply org.gnome.Evince.json:199]: The next release of Evince will use the next version of the SDK which includes libhandy already.

File: org.gnome.Evince.json
Comment (gpoo) [reply org.gnome.Evince.json:161]: > Ditto. Why is it the development version when they tagged it ? Or maybe they have a different release scheme?

From gspell Readme:

> gspell follows the GNOME release schedule and uses the same versioning scheme:
>
>    https://developer.gnome.org/programming-guidelines/stable/versioning.html.en
>
> For example the 1.7.x versions are unstable, and the 1.8.x versions are stable.

You were changing 1.8.4 (stable) by 1.9.1 (unstable). Please avoid that unless there is a good reason to do so.

File: org.gnome.Evince.json
Comment (gpoo) [reply org.gnome.Evince.json:199]: Indeed. No need to add `libhandy` here.

**Comment by barthalion** [comment]
I disagree with random bumps of random modules, and this is essentially what adding x-data-checker accomplishes here. This is fine if an upstream maintainer decides to do so, but not so much if it's done via drive-through contribution.

**Comment by proletarius101** [comment]
> I disagree with random bumps of random modules, and this is essentially what adding x-data-checker accomplishes here. This is fine if an upstream maintainer decides to do so, but not so much if it's done via drive-through contribution.

Well, then what's the upstream's opinion? At least this repo isn't updated in time. And what drives me to make this PR is there were some bugs in the old versions, and it takes so long for the "upstream" to distribute it here.

File: org.gnome.Evince.json
Comment (proletarius101) [reply org.gnome.Evince.json:199]: > The next release of Evince will use the next version of the SDK which includes libhandy already.

That's a good news though. Not sure how long will it take.

**Comment by proletarius101** [comment]
> Odd minor releases are development versions. 3.39.X releases are all development releases for version 3.40 (which will actually be called version 40). This isn't a stable version.

@hadess And I observed that it's already bumped to 3.38.2 (3.38.x). Is that intended? Although I welcome that as [many distros](https://pkgs.org/download/evince) (e.g. Fedora, OpenSUSE) update to the minor increment version.

**Comment by hadess** [comment]
> @hadess And I observed that it's already bumped to 3.38.2 (3.38.x). Is that intended? Although I welcome that as [many distros](https://pkgs.org/download/evince) (e.g. Fedora, OpenSUSE, Debian) update to the minor increment version.

`major.minor.micro`

**Comment by gpoo** [comment]
Thanks for working on this. I applied a reduced patch of this (only for freedesktop and sourceforge). See commit 1a09ea16872895808fe33e222619eb9b7b1bfbc0

**Pull request closed without merging**
